### PR TITLE
Add new config command

### PR DESCRIPTION
**Pull request by nmuesch**
### What does this PR do?

Adds a new `config` root command to the apigentools CLI that will scan the spec config and just list information about the spec being worked on. These currently include the supported languages, api versions, as well as the path to the full yaml spec thats generated.  (Language and version also come with flags that only print supported languages or only supported versions. These also support the global language and version flags if you only care about seeing information for `java` for example)

(Also some light cleanup of unused imports + alphabetizing)

### Description of the Change

The new command will both return, and log the available languages for the given spec. This is to help avoid any confusion about what options are available when running commands like `generate`.

### Verification Process

Added test to go alongside this new command as well as verified locally. 

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: apigentools/cli.py [hunk 1]
```diff
@@ -392,6 +392,10 @@ def get_cli_parser():
         action="store_true",
     )
 
+    sp.add_parser(
+        "list-languages", help="Provide a list of the supported langauges for this spec"
```
Comment (jirikuncar) [inline apigentools/cli.py]: Can we do either options `--list-languages` and `--list-versions` or command `list (languages|versions)`?

Comment (bkabrda) [reply apigentools/cli.py]: Or maybe just `--list` that would list language-version combinations and would have `--versions-only` and `--languages-only` mutually exclusive switches?

Comment (nmuesch) [reply apigentools/cli.py]: Hmm, if they are options I'm not sure which command it should be tied to. 
I can make `list-languages` take in like a `--include-versions` and show each version alongside it. WDYT?

Comment (nmuesch) [reply apigentools/cli.py]: I can go that route ^. I may call it something like `apigentools spec-info` though 🙂 If that approach works for you guys I'll start that route

File: apigentools/commands/generate.py [hunk 2]
```diff
@@ -4,13 +4,10 @@
 # Copyright 2019-Present Datadog, Inc.
 import collections
 import copy
-import glob
 import json
 import logging
 import os
 import subprocess
-import tempfile
-from distutils.dir_util import copy_tree
```
Comment (nmuesch) [inline apigentools/commands/generate.py]: unused imports

File: apigentools/commands/__init__.py [hunk 3]
```diff
@@ -4,18 +4,20 @@
...
 from apigentools.commands.templates import TemplatesCommand
 from apigentools.commands.test import TestCommand
 from apigentools.commands.validate import ValidateCommand
 
 all_commands = {
-    "init": InitCommand,
     "generate": GenerateCommand,
+    "init": InitCommand,
+    "list-languages": ListLanguagesCommand,
+    "push": PushCommand,
```
Comment (nmuesch) [inline apigentools/commands/__init__.py]: alphabetize 😉

File: apigentools/cli.py (hunk 1 above)
Comment (bkabrda) [reply apigentools/cli.py]: I think this is a good route to take, however I'm wondering whether `spec-info` is the right thing to call it. It almost feel like `spec-info` is too broad. However, I don't have a better proposal, so I'll leave this point open to discussion :)

Comment (nmuesch) [reply apigentools/cli.py]: I left it as list-languages, but I've addressed the other review from this thread + used the utility method we have in the base Command class. So if we're ok with `list-languages` I think this is good to go.

File: apigentools/commands/list_config.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        cmd.config = Config.from_file(
+            os.path.join(ctx.get("config_dir"), constants.DEFAULT_CONFIG_FILE)
+        )
+        cmd.run()
+
+
+class ConfigCommand(Command):
+    def run(self):
+        # Yields tuples (language, version, spec_path)
+        language_info = self.yield_lang_version_specfile()
```
Comment (jirikuncar) [inline apigentools/commands/list_config.py]: 👍

**Pull request merged**